src/SystemState/ConnectedState.cpp

```diff
@@ -185,42 +185,41 @@ void ConnectedState::exit(StateMachine& sm)
 
 void ConnectedState::initHttpClient()
 {
-    m_client.regOnResponse([](const HttpResponse& rsp){
+    m_client.regOnResponse([](const int userData, const HttpResponse& rsp) {
         uint16_t statusCode = rsp.getStatusCode();
 
         if (HttpStatus::STATUS_CODE_OK == statusCode)
         {
             LOG_INFO("Online state reported.");
         }
-
     });
 
-    m_client.regOnError([]() {
+    m_client.regOnError([](const int userData) {
         LOG_WARNING("Connection error happened.");

```

Comment:
If the method argument is not used, please use the UTIL_NOT_USED() macro if available. Otherwise use (void)userData.